### PR TITLE
Display Control implementation

### PR DESCRIPTION
**Pull request by alex-deaconu**
Require 'da' role for editing Display Control
Merged player pro modals

Remove display control checkbox
Show subscription modal when Configure button is clicked
when the Display is not rise player pro

Refactoring:
Separate factory for player pro features
Separate screenshot service and directive
Controller flags set on Display object
To allow re-use in other places

### CODE REVIEW

**Comment by alex-deaconu** [comment]
This is the initial PR that implements the UX functionality. Will open subsequent PRs to this one for the Player Pro settings functionality.

@Rise-Vision/apps please review this initial one. Thanks!

File: web/scripts/displays/services/svc-player-pro-factory.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        var displayPlayerVersion = display && parsePlayerDate(display.playerVersion);
+        var minimumVersion = _latestPlayerVersion && (new Date()).setMonth(_latestPlayerVersion.getMonth() - 1);
+        var upToDate = displayPlayerVersion && minimumVersion && displayPlayerVersion >= minimumVersion;
+
+        return !factory.is3rdPartyPlayer(display) &&
+          !factory.isUnsupportedPlayer(display) &&
+          (display && display.playerName && (display.playerName !== 'RisePlayerElectron' || !upToDate));
+      };
+
+      factory.isProCompatiblePlayer = function (display) {
```
Comment (ezequielc) [inline web/scripts/displays/services/svc-player-pro-factory.js]: @alex-deaconu I think we should rename this to isOfflinePlayCompatiblePayer() to make it more clear. What do you think? Thanks

**Comment by alex-deaconu** [comment]
@ezequielc thanks for the review. Good points all around, latest commit should address them. Thanks!

**Comment by ezequielc** [comment]
@alex-deaconu Looks good. Player Pro pop up works, but it still shows 'Try Professional' in the Display row where it should be 'Subscribe'. Inside the popup it says Subscribe to RPP as expected. Thanks!

**Comment by alex-deaconu** [comment]
@ezequielc I tried to work on the issue above with the list saying "Try Professional". Having completely misunderstood what was going on, I fixed something else (will put that fix through shortly).

However, the issue at hand that you are bringing up is an existing issue. The Store API does not return the Trial Availability. Just the subscription status and trial duration. Hence, we have no way of telling if a trial is available, we need to use the other API for that.

@fjvallarino FYI. Any ideas, or am I still misunderstanding this? Thanks.

**Comment by alex-deaconu** [comment]
Just a bit more research on the above. 2 companies, and 2 displays from those companies.

Rise Vision Test Co.
https://store-stage-2.risevision.com/product/2048/rise-player-professional?cid=17899fe3-db05-4ecd-ade4-a7106fe53784
Company Status:
https://store-dot-rvacore-test.appspot.com/v1/company/b428b4e8-c8b9-41d5-8a10-b4193c789443/product/status?pc=c4b368be86245bf9501baaa6e0b00df9719869fd
Display Status:
https://store-dot-rvacore-test.appspot.com/v1/product/c4b368be86245bf9501baaa6e0b00df9719869fd/status?displayIds=NWGFXPAFT24Q

KC Uptime:
https://store-stage-2.risevision.com/product/2048/rise-player-professional?cid=1e08e1e7-f3eb-4de9-bb38-8fa2eb0f7198
Company Status:
https://store-dot-rvacore-test.appspot.com/v1/company/1e08e1e7-f3eb-4de9-bb38-8fa2eb0f7198/product/status?pc=c4b368be86245bf9501baaa6e0b00df9719869fd
Display Status:
https://store-dot-rvacore-test.appspot.com/v1/product/c4b368be86245bf9501baaa6e0b00df9719869fd/status?displayIds=MJ8S5X9VN2JE

For comparison, an On Trial Display Status:
https://store-dot-rvacore-test.appspot.com/v1/product/c4b368be86245bf9501baaa6e0b00df9719869fd/status?displayIds=37EXFCCRXPBJ

I'm still looking for a Company with an expired trial for Rise Player Pro..

**Comment by ezequielc** [comment]
@alex-deaconu I am wondering if this is actually a Store issue. We should be able to check if `trialPeriod` is not 0 to know if trial is available. It works for Company status but I see it shows 30 for the Display, where it should be 0.
As it is an existing issue, and the popup actually shows the correct 'Subscribe' button, I am good with releasing as is and we can fix it later, maybe in Store. Thoughts? Thanks

**Comment by alex-deaconu** [comment]
@ezequielc not sure what the exact source of the problem is, but I agree we can defer and fix later. Thanks!

**Comment by alex-deaconu** [comment]
@ezequielc @fjvallarino my plan is to merge this PR into the `beta/display-control` branch, and then from there open another PR to merge to master. Does that flow work for you? Thanks.

**Comment by fjvallarino** [comment]
@alex-deaconu makes sense. It's cool that they finally added the option of changing the destination branch for a PR!

**Comment by ezequielc** [comment]
@alex-deaconu Sounds good. Thanks!

**Pull request merged**